### PR TITLE
wgengine: use Go API rather than UAPI for status

### PR DESCRIPTION
**Pull request by raggi**
Signed-off-by: James Tucker <james@tailscale.com>

### CODE REVIEW

**Comment by danderson** [comment]
We switched _from_ this to uapi at some point in the past. Anyone remember why we did? I'm worried there's a chesterton's fence lurking here.

**Comment by bradfitz** [comment]
> We switched from this to uapi at some point in the past. Anyone remember why we did? I'm worried there's a chesterton's fence lurking here.

We switched away from this when we switched away from our own wireguard-go fork to upstream wireguard-go.

The idea at the time was that UAPI was slow but we could probably make it faster, which we did, over a number of rounds. But I think we made it about as fast as we can.

**Comment by raggi** [comment]
> @raggi, want to rebase this on the `wgengine/wgint` package for now instead?

done!

File: wgengine/userspace.go [hunk 1]
```diff
@@ -990,126 +989,42 @@ func (e *userspaceEngine) getStatus() (*Status, error) {
...
+		if peer == nil {
+			continue
 		}
+
+		var p ipnstate.PeerStatusLite
+		p.NodeKey = key
+		p.RxBytes = int64(wgint.PeerRxBytes(peer))
+		p.TxBytes = int64(wgint.PeerTxBytes(peer))
+		p.LastHandshake = time.Unix(0, wgint.PeerLastHandshakeNano(peer))
+		peers = append(peers, p)
```
Comment (bradfitz) [inline wgengine/userspace.go:1027]: I'd probably just

```go
peers = append(peers, ipnstate.PeerStatusLite{
     NodeKey: ...
     RxBytes: ...
})
```

but either way

**Review by bradfitz — APPROVED** [review]

File: wgengine/userspace.go (hunk 1 above)
Comment (raggi) [reply wgengine/userspace.go:1027]: I'll apply that in the second PR when the code moves into the per-peer getter.

Comment (raggi) [reply wgengine/userspace.go:1027]: Oh, actually, I'm using the return var there, so 🤷🏼 :)

**Pull request merged**